### PR TITLE
Simplify Google Cloud credential detection

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
There is no need of duplicating the behavior of the [`google.FindDefaultCredentials`](https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials) function.

The only custom credential source we introduce is `GOOGLE_APPLICATION_CREDENTIALS_DATA` and only because it's not supported yet:

* https://github.com/googleapis/google-api-go-client/issues/185

### CODE REVIEW

File: iterative/testdata/script_template_cloud_gcp.golden
Comment (0x2b3bfa0) [inline iterative/testdata/script_template_cloud_gcp.golden:43]: Leaving blank because `7 value with "quotes" and spaces` is not a valid credentials JSON object and `CredentialsFromJSON` can't parse it.

**Review by DavidGOrtega — APPROVED** [review]
👍

**Comment by 0x2b3bfa0** [comment]
Waiting for @dacbd approval before merging, because this cascades on his pull request.

**Review by dacbd — APPROVED** [review]

**Pull request merged**
